lib/ui/cachedblock.js

```diff
@@ -0,0 +1,55 @@
+const Block = require('./block');
+const { blockSize } = require('./variables');
+
+class CachedBlock extends Block {
```

Comment:
But this caching stuff doesn't belong there. It might actually break things with a separate DOM engine.